File: java-source/src/test/java/net/corda/accord/flow/PromissoryNoteSettleFlowTests.java
package net.corda.accord.flow;

import net.corda.testing.node.*;

import static net.corda.testing.node.NodeTestUtils.ledger;

/**
 * TODO: Write tests for promissory note settling once functionality is complete.
 */
public class PromissoryNoteSettleFlowTests {

    private MockNetwork mockNetwork;
    private StartedMockNode a, b, c;

//    @Before
//    public void setup() {
//
//        MockNetworkParameters mockNetworkParameters = new MockNetworkParameters().withNotarySpecs(Arrays.asList(new MockNetworkNotarySpec(new CordaX500Name("Notary", "London", "GB"))));
//        mockNetwork = new MockNetwork(Arrays.asList("net.corda.accord", "net.corda.finance.contracts.asset", "net.corda.finance.schemas"), mockNetworkParameters);
//        System.out.println(mockNetwork);
//
//        a = mockNetwork.createNode(new MockNodeParameters());
//        b = mockNetwork.createNode(new MockNodeParameters());
//        c = mockNetwork.createNode(new MockNodeParameters());
//
//        ArrayList<StartedMockNode> startedNodes = new ArrayList<>();
//        startedNodes.add(a);
//        startedNodes.add(b);
//        startedNodes.add(c);
//
//        // For real nodes this happens automatically, but we have to manually register the flow for tests
//        startedNodes.forEach(el -> el.registerInitiatedFlow(PromissoryNoteSettleFlow.Responder.class));
//        mockNetwork.runNetwork();
//    }
//
//    @After
//    public void tearDown() {
//        mockNetwork.stopNodes();
//    }
//
//    @Rule
//    public final ExpectedException exception = ExpectedException.none();

}

File: kotlin-source/src/test/kotlin/net/corda/accord/flow/PromissoryNoteSettleFlowTests.kt
package net.corda.accord.flow

import net.corda.core.identity.CordaX500Name
import net.corda.core.internal.packageName
import net.corda.finance.schemas.CashSchemaV1
import net.corda.testing.node.MockNetwork
import net.corda.testing.node.MockNetworkNotarySpec
import net.corda.testing.node.MockNodeParameters
import net.corda.testing.node.StartedMockNode
import org.junit.After
import org.junit.Before

/**
 * Practical exercise instructions Flows part 3.
 * Uncomment the unit tests and use the hints + unit test body to complete the FLows such that the unit tests pass.
 */
class PromissoryNoteSettleFlowTests {
    lateinit var mockNetwork: MockNetwork
    lateinit var a: StartedMockNode
    lateinit var b: StartedMockNode
    lateinit var c: StartedMockNode

    @Before
    fun setup() {
        mockNetwork = MockNetwork(listOf("net.corda.accord", "net.corda.finance.contracts.asset", CashSchemaV1::class.packageName),
                notarySpecs = listOf(MockNetworkNotarySpec(CordaX500Name("Notary","London","GB"))))
        a = mockNetwork.createNode(MockNodeParameters())
        b = mockNetwork.createNode(MockNodeParameters())
        c = mockNetwork.createNode(MockNodeParameters())
        val startedNodes = arrayListOf(a, b, c)
        // For real nodes this happens automatically, but we have to manually register the flow for tests
        startedNodes.forEach { it.registerInitiatedFlow(PromissoryNoteIssueResponderFlow::class.java) }
        startedNodes.forEach { it.registerInitiatedFlow(PromissoryNoteSettleFlowResponder::class.java) }
        mockNetwork.runNetwork()
    }

    @After
    fun tearDown() {
        mockNetwork.stopNodes()
    }
}
